### PR TITLE
Fix wcs.array_shape  definition in WcsGeom.create

### PR DESCRIPTION
**Pull request by QRemy**
Fix wcs.array_shape  definition in WcsGeom.create to fix the bug described in :
https://github.com/gammapy/gammapy/pull/4652#issuecomment-1645298025

### CODE REVIEW

**Comment by Astro-Kirsty** [comment]
Thanks @QRemy. 
Are the tests failing because of the known CI issue, or because this reverses the order where we have no implemented that before?

**Review by Astro-Kirsty — COMMENTED** [review]
Thanks @QRemy.
Are the tests failing because of the known CI issue, or because this reverses the order where we have no implemented that before?

**Comment by registerrier** [comment]
Thanks @QRemy . I am afraid the problems run deeper. 

For instance, the `get_shape` function used in `WcsGeom.create` handles differently tuples and lists.

See:
```
>>> print(WcsGeom.create(binsz=(0.1,0.05)).wcs)
WCS Keywords

Number of WCS axes: 2
CTYPE : 'RA---CAR'  'DEC--CAR'  
CRVAL : 0.0  0.0  
CRPIX : 1800.5  1800.5  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : -0.1  0.05  
NAXIS : 3600  3600
```

but:
```
>>> print(WcsGeom.create(binsz=[0.1,0.05]).wcs)
WCS Keywords

Number of WCS axes: 2
CTYPE : 'RA---CAR'  'DEC--CAR'  
CRVAL : 0.0  0.0  
CRPIX : 1800.5  900.5  
PC1_1 PC1_2  : 1.0  0.0  
PC2_1 PC2_2  : 0.0  1.0  
CDELT : -0.1  0.1  
NAXIS : 1800  3600
```

**Comment by registerrier** [comment]
According to https://astropy-astrofrog.readthedocs.io/en/latest/api/astropy.wcs.WCS.html#astropy.wcs.WCS.array_shape

the expected order is (rows, columns). It should then be reversed as you propose.

File: gammapy/maps/wcs/geom.py [hunk 1]
```diff
@@ -88,12 +88,18 @@ class WcsGeom(Geom):
...
         self._wcs = wcs
         self._frame = wcs_to_celestial_frame(wcs).name
         self._projection = wcs.wcs.ctype[0][5:]
         self._axes = MapAxes.from_default(axes, n_spatial_axes=2)
 
+        if npix is None:
+            if wcs.array_shape is not None:
+                npix = wcs.array_shape[::-1]
+            else:
+                npix = (0, 0)
```
Comment (registerrier) [inline gammapy/maps/wcs/geom.py:101]: Can this be tested?

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy. Looks good. See few inline comments.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy . Merging now.

**Pull request merged**